File: Cpp_Algo/백준/Gold/1068. 트리/트리.cc
#include <iostream>
using namespace std;

int N, K;
int arr[51];
int answer = 0;

void dfs(int node) {
	short is_leaf = 1;

	if (node == K) {
		return;
	}

	for (int i = 0; i < N; i++)
	{
		if (arr[i] == node && i != K) {
			dfs(i);
			is_leaf = 0;
		}
	}

	if (is_leaf == 1) {
		answer++;
	}
}

int main()
{
	cin >> N;

	for (int i = 0; i < N; i++)
	{
		cin >> arr[i];
	}

	cin >> K;

	for (int i = 0; i < N; i++)
	{
		if (arr[i] == -1) {
			dfs(i);
		}
	}

	cout << answer;
}

File: 백준/Gold/1068. 트리/트리.py
def dfs(node):
    global answer

    if node == K:
        return

    is_leaf = True

    for next_node in range(N):
        if arr[next_node] == node and not next_node == K:
            dfs(next_node)
            is_leaf = False

    if is_leaf:
        answer += 1


N = int(input())
arr = list(map(int, input().split()))
K = int(input())

answer = 0

root = 0

for i in range(N):
    if arr[i] == -1:
        root = i

dfs(root)
print(answer)
